src/Look/InfoBoxLook.cpp

```diff
@@ -73,10 +75,11 @@ InfoBoxLook::Initialise(bool _inverse, bool use_colors,
 }
 
 void
-InfoBoxLook::ReinitialiseLayout(unsigned width)
+InfoBoxLook::ReinitialiseLayout(unsigned width, unsigned scale_title_font)
 {
   FontDescription title_font_d(8);
-  AutoSizeFont(title_font_d, width, _T("123456789012345"));
+  unsigned scaled_width = width * Clamp((int)scale_title_font,50,150) / 100;
```

Comment:
Be careful, as you might be doing an integer division here. 150 / 100 == 1

Reply:
More dangerously, `50/100 == 0` - what happens when `scaled_width==0`?

Reply:
It is probably fine here though, since multiplication by width goes first.

Reply:
True, if width == 1.
Arithmetic expressions are evaluated left to right. I could add parentheses, though. Are you suggesting this?

Reply:
No, @bomilkar you're right, this expression is ok after all. Truncation happens at the end.
What is questionable however is the cast to `int` - why did you do that?

Reply:
Clamp wants either 3 int **or** 3 unsigned.
Clamp((int)scale_title_font,50,150)
Clamp(scale_title_font,(unsigned)50,(unsigned)150)

Reply:
That's what I thought. You're using the wrong integer literals, and you used a cast to the wrong type to work around that problem.
How about:
```
Clamp(scale_title_font, 50U, 150U)
```
It's pointless to use signed literals and then cast them to unsigned. Just use unsigned literals.
Even though an overflow during the cast is unlikely, it's always better to use the correct data types instead of casting forth and back.

Reply:
Looks good! Learned something.
Will be in the next commit .